test/e2e/kubevirt.go

```diff
@@ -1064,15 +1068,39 @@ if [ "$ipv6" != "" ]; then
 	iperf3 -s -D --bind $ipv6 --logfile /tmp/test_${ipv6}_iperf3.log
 fi
 `
-		createIperfServerPods = func(nodes []corev1.Node, netConfig networkAttachmentConfig) ([]*corev1.Pod, error) {
-			annotations := map[string]string{}
-			if netConfig.role != "primary" {
-				annotations["k8s.v1.cni.cncf.io/networks"] = fmt.Sprintf(`[{"name": %q}]`, netConfig.name)
+		nextIPs = func(idx int, subnets []string) ([]string, error) {
+			var ips []string
+			for _, subnet := range subnets {
+				ip, ipNet, err := net.ParseCIDR(subnet)
+				if err != nil {
+					return nil, err
+				}
+				for _ = range idx {
+					ip = iputils.NextIP(ip)
+				}
+				ipNet.IP = ip
+				ips = append(ips, ipNet.String())
 			}
+			return ips, nil
+		}
+		createIperfServerPods = func(nodes []corev1.Node, netConfig networkAttachmentConfig, staticSubnets []string) ([]*corev1.Pod, error) {
 			var pods []*corev1.Pod
-			for _, node := range nodes {
+			for i, node := range nodes {
+				var nse *nadapi.NetworkSelectionElement
+				if netConfig.role != "primary" {
```

Comment:
NIT: maybe better say directly `if netConfig.role == "secondary" {`

Reply:
Problem is that empty role is secondary too, so I think this is better than doing
```golang
if netConfig.role == "" || netConfig.role == "secondary"
```